File: modules/calendar-bridge/android/src/main/java/expo/modules/calendarbridge/CalendarBridgeModule.kt
package expo.modules.calendarbridge

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

import expo.modules.calendarbridge.CalendarDate

class CalendarBridgeModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("CalendarBridge")

        Property("todayDate") {
            CalendarDate.todayDate
        }

        Function("getMonthProps") {
            calendar: String, month: Int, year: Int -> CalendarDate
            .getMonthProps(calendar, month, year)
        }

        Function("getHijrahDate") {
            day: Int, month: Int, year: Int -> CalendarDate
            .getHijrahDate(day, month, year)
        }

        Function("getGregorianDate") {
            day: Int, month: Int, year: Int -> CalendarDate
            .getGregorianDate(day, month, year)
        }

        Function("convertHijriToGregorian") {
            day: Int, month: Int, year: Int -> CalendarDate.convertHijriToGregorian(
                day, month, year
            )
        }

        Function("convertGregorianToHijri") {
            day: Int, month: Int, year: Int -> CalendarDate.convertGregorianToHijri(
                day, month, year
            )
        }
    }
}

File: modules/calendar-bridge/src/CalendarBridgeModule.ts
import { requireNativeModule } from "expo";

import { CalendarType, DateType, MonthPropsType } from "./CalendarBridge.types";

declare class CalendarBridgeModule {
  todayDate: DateType;

  getMonthProps(
    calendar: CalendarType,
    month: number,
    year: number,
  ): MonthPropsType;
  getHijrahDate(day: number, month: number, year: number): DateType;
  getGregorianDate(day: number, month: number, year: number): DateType;
  convertGregorianToHijri(day: number, month: number, year: number): DateType;
  convertHijriToGregorian(day: number, month: number, year: number): DateType;
  adjustDay(newNumber: number): void;
}

export default requireNativeModule<CalendarBridgeModule>("CalendarBridge");
